src/wp-admin/includes/image.php

```diff
@@ -384,30 +504,43 @@ function wp_create_image_subsizes( $file, $attachment_id ) {
  * Errors are stored in the returned image metadata array.
  *
  * @since 5.3.0
+ * @since 6.1.0 The $mime_type parameter was added.
  * @access private
  *
- * @param array  $new_sizes     Array defining what sizes to create.
- * @param string $file          Full path to the image file.
- * @param array  $image_meta    The attachment meta data array.
- * @param int    $attachment_id Attachment ID to process.
+ * @param array  $new_sizes       Array defining what sizes to create.
+ * @param string $file            Full path to the image file.
+ * @param array  $image_meta      The attachment meta data array.
+ * @param int    $attachment_id   Attachment ID to process.
+ * @param string $mime_type       Optional. The mime type to check for missing sizes. Default is the image mime of $file.
  * @return array The attachment meta data with updated `sizes` array. Includes an array of errors encountered while resizing.
  */
-function _wp_make_subsizes( $new_sizes, $file, $image_meta, $attachment_id ) {
+function _wp_make_subsizes( $new_sizes, $file, $image_meta, $attachment_id, $mime_type = '' ) {
 	if ( empty( $image_meta ) || ! is_array( $image_meta ) ) {
 		// Not an image attachment.
 		return array();
 	}
 
+	$sizes = array();
+
+	$original_mime_type = wp_get_image_mime( $file );
+	if ( ! $mime_type ) {
+		$mime_type = $original_mime_type;
+	}
+
 	// Check if any of the new sizes already exist.
-	if ( isset( $image_meta['sizes'] ) && is_array( $image_meta['sizes'] ) ) {
-		foreach ( $image_meta['sizes'] as $size_name => $size_meta ) {
+	if ( isset( $sizes ) && is_array( $sizes ) ) {
```

Comment:
I may be missing something here, but thought I'd check in.
It looks like this is checking `$sizes`, which is set to `array()` just above and can only be empty.

Why is `$image_meta['sizes']` no longer checked?

I haven't made it through the rest of the PR yet, so it's possible I'm missing context.
If this is no longer needed due to other flow changes, would it make sense to change things, since the variable is set to `array()` just above?

Reply:
Good catch! this was a bug probably from a late refactor that would only affect retries (when `_wp_make_subsizes` is used. I restored the previous behavior now (the only difference from trunk now is adding the mime type checks so each mime type can be checked.

I verified retries work by setting my PHP timeout very low and uploading a large image.